src/D2L.CodeStyle.TestAnalyzers/SourceAttribute/TestCaseSourceAttributeAnalyzer.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace D2L.CodeStyle.TestAnalyzers.SourceAttribute {
+	[DiagnosticAnalyzer( LanguageNames.CSharp )]
+	class TestCaseSourceAttributeAnalyzer : DiagnosticAnalyzer {
+		public const string DiagnosticId = "D2L0007";
+		private const string Category = "Safety";
+
+		private const string Title = "Ensure the source of 'TestCaseSourceAttribute' is static field, property or method in tests.";
+		private const string Description = "The source of 'TestCaseSourceAttribute' must be static field, property or method in tests.";
+		internal const string MessageFormat = "Add 'static' for the source {0} of 'TestCaseSourceAttribute' for NUnit 3 compatibility";
+
+		private static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+			DiagnosticId,
+			Title,
+			MessageFormat,
+			Category,
+			DiagnosticSeverity.Warning,
+			isEnabledByDefault: true,
+			description: Description
+		);
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create( Rule );
+
+		public override void Initialize( AnalysisContext context ) {
+			context.RegisterSemanticModelAction(

```

Comment:
When does this action typically run? 
I'm curious if registering an attribute declaration syntax action might be better? Find all usages of the attribute, pull out the thing being referenced by the source (can do something like `context.GetSymbolInfo( syntaxNode ).Symbol`), and verify that it's static. 
Thoughts?